agent/consul/leader_routine_manager.go

```diff
@@ -23,7 +25,13 @@ type LeaderRoutineManager struct {
 
 func NewLeaderRoutineManager(logger *log.Logger) *LeaderRoutineManager {
 	if logger == nil {
-		logger = log.New(os.Stderr, "", log.LstdFlags)
+		consulLogger := hclog.New(&hclog.LoggerOptions{
```

Comment:
Repeat of the github.com/hashicorp/consul/lib helper function idea from earlier.